subprojects/problems/src/main/java/org/gradle/api/problems/internal/GradleExceptionWithProblem.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.problems.internal;
+
+import org.gradle.api.problems.interfaces.Problem;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Represents some chunk of work.
+ *
+ * @since 8.3
+ */
+public class GradleExceptionWithProblem extends RuntimeException {
```

Comment:
🤔 We should use a plural name, as this exception can contain multiple problems.

```suggestion
public class GradleExceptionWithProblems extends RuntimeException {
```